realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/ProgressListenerTests.kt

```diff
@@ -325,30 +322,30 @@ class ProgressListenerTests {
 
     // Create remote data for a given user.
     private fun createRemoteData(config: SyncConfiguration) {
-        val realm = Realm.getInstance(config)
-        val changesUploaded = CountDownLatch(1)
-        writeSampleData(realm)
-        val session: SyncSession = realm.syncSession
-        session.addUploadProgressListener(ProgressMode.CURRENT_CHANGES, object : ProgressListener {
-            override fun onChange(progress: Progress) {
-                if (progress.isTransferComplete) {
-                    session.removeProgressListener(this)
-                    changesUploaded.countDown()
+        Realm.getInstance(config).use { realm ->
+            val changesUploaded = CountDownLatch(1)
+            val session: SyncSession = realm.syncSession
+            writeSampleData(realm)
+            session.addUploadProgressListener(ProgressMode.CURRENT_CHANGES, object : ProgressListener {
+                override fun onChange(progress: Progress) {
+                    if (progress.isTransferComplete) {
+                        session.removeProgressListener(this)
+                        changesUploaded.countDown()
+                    }
                 }
-            }
-        })
-        TestHelper.awaitOrFail(changesUploaded)
-        realm.close()
+            })
+            TestHelper.awaitOrFail(changesUploaded)
+        }
     }
 
     private fun getStoreTestDataSize(config: RealmConfiguration): Long {
-        val realm: Realm = Realm.getInstance(config)
-        val objectCounts: Long = realm.where<SyncDog>().count()
-        realm.close()
-        return objectCounts
+        Realm.getInstance(config).use { realm ->
+            realm.refresh()
```

Comment:
Hmm, is refreshing here really neeeded?

Reply:
Not really. I had some troubles getting my head around the semantics of the progress listeners from the start. I reverted the change. 

Conceptually it could make sense to have here as the check would most likely always make sense to do after refreshing, but due to the heavy reliance on order of operations the test are already written with the order in mind.